File: Recursion/Print 1 To N Without Loop/PrintWithoutLoop.java
import java.util.Scanner;

class Solution {
  public void printNos(int n) {
    if (n == 0) return;
    printNos(n - 1);
    System.out.print(n + " ");
  }
}

class PrintWithoutLoop {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int T = sc.nextInt();
    while (T-- > 0) {
      Solution obj = new Solution();
      int N;
      N = sc.nextInt();
      obj.printNos(N);
      System.out.println();
      System.out.println("~");
    }
    sc.close();
  }
}

File: Recursion/Print 1 To N Without Loop/printWithoutLoop.cpp
#include <iostream>
using namespace std;
class Solution {
 public:
  void printNos(int n) {
    if (n == 0) return;
    printNos(n - 1);
    cout << n << " ";
  }
};

int main() {
  int T;
  cin >> T;
  while (T--) {
    int N;
    cin >> N;
    Solution ob;
    ob.printNos(N);
    cout << "\n";
    cout << "~"
         << "\n";
  }
  return 0;
}